.github/workflows/run-testing.yaml

```diff
@@ -26,105 +28,66 @@ env:
           fetch-depth: 0
 
-      - name: Install Kubernetes tools
-        uses: yokawasa/action-setup-kube-tools@v0.11.2
+      - name: Restore binaries from cache
+        uses: actions/cache/restore@v4
         with:
-          setup-tools: |
-            helmv3
-            kubectl
-          helm: "${{ env.HELM_VERSION }}"
-          kubectl: "${{ matrix.k8s_version }}"
+          path: ./bin
+          key: binary
 
-      - uses: actions/setup-python@v5
+      - name: Setup Go
+        uses: actions/setup-go@v5
         with:
-          python-version: "3.13"
+          go-version-file: "test/go.mod"
+          check-latest: true
+          cache: true
+        id: go
 
-      - name: Set up chart-testing
-        uses: helm/chart-testing-action@v2.7.0
-        with:
-          version: 3.13.0
+      - name: Install go-junit-report
+        run: |
+          # This is a fork, so go install won't work
+          git clone https://github.com/joschi/go-junit-report
+          cd go-junit-report
+          go install ./cmd/...
 
       - name: Create Kind cluster
         uses: helm/kind-action@v1
         with:
           node_image: kindest/node:${{ matrix.k8s_version }}
-          kubectl_version: ${{ matrix.k8s_version }}
 
-      - name: Remove node taints
-        run: |
-          kubectl taint --all=true nodes node.cloudprovider.kubernetes.io/uninitialized- || true
+      - name: Add helm repo dependencies
+        run: yq -M eval '.repositories[] | (.name + " " + .url)' .github/ci/helm-repos.yaml | xargs -L 1 helm repo add
 
-      - name: Run chart-testing (install)
-        env:
-          LICENSE_KEY: ${{ secrets.ENTERPRISE_COMPONENTS_LICENSE_KEY }}
+      - name: Run tests
         run: |
-          kubectl create namespace ${{ matrix.chart }}-testing
-          kubectl create secret -n=${{ matrix.chart }}-testing generic license --from-literal=key=$LICENSE_KEY
-          helm dep update charts/${{ matrix.chart }}
-          CRDS=$(helm show crds charts/${{ matrix.chart }})
-          if [[ -n "$CRDS" ]]; then
-            echo "$CRDS" | kubectl apply -f -
-          fi
-          if [ -f charts/${{ matrix.chart }}/e2e/simple.yaml ]; then
-            helm template charts/${{ matrix.chart }} \
-              --namespace ${{ matrix.chart }}-testing \
-              --values charts/${{ matrix.chart }}/e2e/simple.yaml \
-              --validate | kubectl apply --server-side -f - --dry-run=server
-            cd charts/${{ matrix.chart }}
-            ct install \
-              --release-name test \
-              --namespace ${{ matrix.chart }}-testing \
-              --config ../../.github/ci/ct.yaml \
-              --charts . \
-              --helm-extra-set-args "--values e2e/simple.yaml" --skip-clean-up
-          else
-            echo "skipping tests since to e2e values were found"
-          fi
+          cd test && go test -json ./... 2>&1 | go-junit-report -set-exit-code -parser gojson -iocopy -out ../report.xml
```

Comment:
have you tried running several threads?

Reply:
It would make a mess due to CRDs. Some tests don't need it, some install and manage it via Helm and some use operator to do that. This is why test don't set `t.Parallel()` after recent changes

The safest option here it to run each test separately. Added `-p 1` just to be sure